### PR TITLE
arch: arm: Set Zero Latency IRQ to priority level zero:

### PR DESCRIPTION
**Pull request by joerchan**
Set Zero Latency IRQ to priority level zero and SVCs to priority level
one when Zero Latency IRQ is enabled.
This makes Zero Zatency truly zero latency when the kernel has been
configured with userspace enabled, or when IRQ offloading is used.

Exceptions can still delay Zero Latency IRQ, but this is considered
ok since exceptions indicate a serious error, and the system needs to
recover.

Fixes: #7869

### CODE REVIEW

**Comment by joerchan** [comment]
FYI: @carlescufi  @ioannisg  @andrewboie

**Comment by ioannisg** [comment]
Are there any implications from SVC being blocked by zero-latency peripheral IRQs?

I agree that Fault exceptions having highest priority (equal to zero-latency) is OK.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: include/arch/arm/cortex_m/exc.h [hunk 1]
```diff
@@ -27,12 +27,13 @@ extern "C" {
 #define _EXCEPTION_RESERVED_PRIO 0
 #endif
 
-#define _EXC_SVC_PRIO 0
 #define _EXC_FAULT_PRIO 0
 #ifdef CONFIG_ZERO_LATENCY_IRQS
 #define _EXC_ZERO_LATENCY_IRQS_PRIO 1
+#define _EXC_SVC_PRIO 1
```
[the remark this answers is not in the document]
Comment (joerchan) [reply include/arch/arm/cortex_m/exc.h]: I don't think SVCs should be masked, they weren't before, and shouldn't be now.
ZLI at the same priority as SVC, I messed up in the rebase here, it should be 0.

**Comment by carlescufi** [comment]
@cvinayak can you revisit? This looks good to me but I'd like someone else to take a look.
@agross-linaro, @andrewboie, @andyross any comments on this one?

**Review by cvinayak — APPROVED** [review]
LGTM

**Review by ioannisg — APPROVED** [review]

**Review by andyross — APPROVED** [review]
I'm not expert enough on ARM Cortex priority handling to review the design, but the logic looks correct and the documentation and justification are clear.

**Comment by carlescufi** [comment]
@andyross thank you

**Pull request merged**